xtrlock.c

```diff
@@ -36,38 +36,102 @@
 #include <math.h>
 #include <ctype.h>
 #include <values.h>
+#include <unistd.h>
+#include <stdbool.h>
 
 #ifdef SHADOW_PWD
 #include <shadow.h>
 #endif
 
+//#define DEBUG
+#ifdef DEBUG
+#define debug_print(...) do{{printf(__VA_ARGS__);} }while (0)
+#else
+#define debug_print(...) do{}while (0)
+#endif
+
 Display *display;
-Window window, root;
+Window window, blank_window, trans_window, root;
 
+//#define DISALLOW_ONE_C_PW
 #define TIMEOUTPERATTEMPT 30000
 #define MAXGOODWILL  (TIMEOUTPERATTEMPT*5)
 #define INITIALGOODWILL MAXGOODWILL
 #define GOODWILLPORTION 0.3
 
+struct {/*Setting correspond to the custom passwd setting. --d0048*/ 
+   bool enable;
+   bool crypt;
+   char* pwd;
+} cust_pw_setting;
+
+bool blank_screen = false;
+int blink_delay = 100000;
```

Comment:
Since these two values may vary according to the arguments passed in, I don't think there is any advantage to define them in macro. What do you think?